src/guide/receive_onchain.md

```diff
@@ -9,47 +9,60 @@ In such cases, the SDK can create a swap and receive the payment without the nee
 <section>
 
 ```rust,ignore
-{{#include ../../snippets/rust/src/receive_onchain.rs:generate-receive-onchain-address}}
+{{#include ../../snippets/rust/src/receive_payment.rs:prepare-receive-payment-onchain}}
+
+{{#include ../../snippets/rust/src/receive_payment.rs:receive-payment}}
 ```
 </section>
 
 <div slot="title">Swift</div>
 <section>
 
 ```swift,ignore
-{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceiveOnchain.swift:generate-receive-onchain-address}}
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift:prepare-receive-payment-onchain}}
+
+{{#include ../../snippets/swift/BreezSDKExamples/Sources/ReceivePayment.swift:receive-payment}}
 ```
 </section>
 
 <div slot="title">Kotlin</div>
 <section>
 
+First, prepare the payment via `prepareReceivePayment`:
+```kotlin,ignore
+{{#include ../../snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ReceivePayment.kt:prepare-receive-payment-onchain}}
+```
+Then execute the payment with the given response:
```

Comment:
Seems to be comments only for Kotlin

Reply:
Yes, since we use `try` statements I figured it would be better to split the snippets with comments in Kotlin's case

Reply:
The more snippets diverge, the more technical debt we accumulate which makes snippet maintenance harder. So IMO it's best to keep the snippets as similar as possible across languages.

Why not make the kotlin snippets start and end just within the try/catch block? There's an implied try/catch in every other language. Then it won't need these extra comments.

Alternatively, why not add these comments universally for all snippets?